client/js/app.js

```diff
@@ -561,6 +554,11 @@
         }
     }
 
+    function contain(min, max, value) {
+        return Math.min(max, Math.max(min, value));
```

Comment:
Is this function returning ```True``` if ```value``` is between ```min``` and ```max```?
If this is wanted you should rename the function.